openapi.yaml

```diff
@@ -1,190 +1,513 @@
+              - restrooms
+              - scenic view point
+              - soccer fields
+              - tennis courts
+              - volleyball
+        - in: query
+          name: parkName
+          description: Filter parks by name
           required: false
           type: string
+        - in: query
+          name: ownerId
+          description: Filter parks by owner ID
+          required: false
+          type: string
+        - in: query
+          name: city
+          description: Filter parks by city
+          required: false
+          type: string
+        - in: query
+          name: state
+          description: Filter parks by state
+          required: false
+          type: string
+          pattern: '^(?:(A[KLRZ]|C[AOT]|D[CE]|FL|GA|HI|I[ADLN]|K[SY]|LA|M[ADEINOST]|N[CDEHJMVY]|O[HKR]|P[AR]|RI|S[CD]|T[NX]|UT|V[AIT]|W[AIVY]))$'
+        - in: query
+          name: zipCode
+          description: Filter parks by zip code
+          required: false
+          type: integer
+      responses:
+        '200':
+          description: Successful Response
+          schema:
+            $ref: '#/definitions/ParksResult'
+        '400':
+          $ref: '#/responses/400'
+        '500':
+          $ref: '#/responses/500'
+    post:
+      summary: Creates a new park
+      tags:
+        - parks
+      operationId: createPark
+      parameters:
+      - $ref: '#/parameters/parkPostBody'
+      responses:
+        '201':
+          $ref: '#/responses/201'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          description: Conflict with id or type
+          schema:
+            $ref: '#/definitions/ErrorResult'
+        '500':
+          $ref: '#/responses/500'
+  /parks/{parkId}:
+    get:
+      summary: Lists info for a specific park
+      tags:
+        - parks
+      description: Get park by unique ID
+      operationId: getParkById
+      parameters:
+        - $ref: '#/parameters/parkIdPath'
+      responses:
+        '200':
+          description: Successful Response
+          schema:
+            $ref: '#/definitions/ParkResult'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+    patch:
+      summary: Update information of a specific park
+      tags: 
+        - parks
+      description: Update park by unique ID
+      operationId: updateParkById
+      parameters:
+        - $ref: '#/parameters/parkIdPath'
+        - $ref: '#/parameters/parkPatchBody'
       responses:
         '200':
-          description: Successful response
+          description: Successful Response
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/ParkResult'
         '400':
-          description: Bad request
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          description: Resource conflict
```

Comment:
id and type are required in the body of the patch request, but should not be able to be changed because id is not client generated and there is no other valid type than park. I'll update the description of 409 to state that these members can't be changed.